File: FullstackOverview.Data/Entities/ChannelMessage.cs
using System;

namespace FullstackOverview.Data.Entities
{
    public class ChannelMessage
    {
        public int Id { get; set; }
        public int ChannelId { get; set; }
        public int? UploadId { get; set; }
        public int UserId { get; set; }
        public string Value { get; set; }
        public bool IsUpload { get; set; }
        public DateTime MessageDate { get; set; }

        public Channel Channel { get; set; }
        public Upload Upload { get; set; }
        public User User { get; set; }
    }
}

File: FullstackOverview.Web/ClientApp/src/app/models/channel-message.ts
import { Channel } from './channel';
import { User } from './user';
import { Upload } from './upload';

export class ChannelMessage {
  id: number;
  channelId: number;
  uploadId: number;
  userId: number;
  value: string;
  isUpload: boolean;
  messageDate: Date;

  channel: Channel;
  upload: Upload;
  user: User;
}
